python/ray/tune/examples/xgboost_dynamic_resources_example.py

```diff
@@ -13,13 +13,72 @@
 from ray.tune import Trainable
 from ray.tune.resources import Resources
 from ray.tune.utils.placement_groups import PlacementGroupFactory
-from ray.tune.suggest.basic_variant import BasicVariantGenerator
 from ray.tune.trial import Trial
 from ray.tune import trial_runner
+from ray.tune.integration.xgboost import TuneReportCheckpointCallback
 
+CHECKPOINT_FILENAME = "model.xgb"
 
-# Dynamic resource allocation is currently only possible
-# with Trainable (class) API
+
+def get_best_model_checkpoint(analysis):
+    best_bst = xgb.Booster()
+    try:
+        with open(analysis.best_checkpoint, "rb") as inputFile:
+            _, _, raw_model = pickle.load(inputFile)
+        best_bst.load_model(bytearray(raw_model))
+    except IsADirectoryError:
+        best_bst.load_model(
+            os.path.join(analysis.best_checkpoint, CHECKPOINT_FILENAME))
+    accuracy = 1. - analysis.best_result["eval-logloss"]
+    print(f"Best model parameters: {analysis.best_config}")
+    print(f"Best model total accuracy: {accuracy:.4f}")
+    return best_bst
+
+
+# FUNCTION API EXAMPLE
```

Comment:
can you split this into 2 examples?

Reply:
I wanted to keep it in one file as to make docstrings and refs simpler, but if you feel that it would be better to split it, I can do that @richardliaw